src/sched/randws.c

```diff
@@ -82,8 +82,7 @@ static void sched_run(ABT_sched sched)
         /* Execute one work unit from the scheduler's pool */
         ABT_pool pool = p_pools[0];
         ABTI_pool *p_pool = ABTI_pool_get_ptr(pool);
-        size_t size = ABTI_pool_get_size(p_pool);
-        if (size > 0) {
+        if (num_pools == 1 || ABTI_pool_get_size(p_pool) > 0) {
```

Comment:
same comment as above with a slight variation.
We `pop` a unit, and if that unit is `NULL` then work-steal.